### ISSUE TITLE
Motion sensor Ranger 2C 3MP

### ISSUE THREAD

**Issue by jlmadrugac** [Imou-OpenPlatform/Imou-Home-Assistant#10]
Hello, motion sensor is missing, I have just purchased balance in the api but can't use the motion sensor.
Is there plans to get this working?

Thanks!

**Comment by Imou-OpenPlatform** [comment]
> Hello, motion sensor is missing, I have just purchased balance in the api but can't use the motion sensor. Is there plans to get this working?
> 
> Thanks!


Hello, are you referring to the motion detect switch? Please try reloading the imou_life integration. If the issue persists, please check the Home Assistant logs for any abnormalities and provide them to us.

**Comment by jlmadrugac** [comment]
> > Hello, motion sensor is missing, I have just purchased balance in the api but can't use the motion sensor. Is there plans to get this working?
> > Thanks!
> 
> Hello, are you referring to the motion detect switch? Please try reloading the imou_life integration. If the issue persists, please check the Home Assistant logs for any abnormalities and provide them to us.

No, the switch is working perfect, but I need to have a sensor if "human detected" that is missing

**Comment by Imou-OpenPlatform** [comment]
> > > Hello, motion sensor is missing, I have just purchased balance in the api but can't use the motion sensor. Is there plans to get this working?
> > > Thanks!
> > 
> > 
> > Hello, are you referring to the motion detect switch? Please try reloading the imou_life integration. If the issue persists, please check the Home Assistant logs for any abnormalities and provide them to us.
> 
> No, the switch is working perfect, but I need to have a sensor if "human detected" that is missing

Hello, human detected is currently not supported. We will consider supporting it in later versions

**Comment by jlmadrugac** [comment]
> > > > Hello, motion sensor is missing, I have just purchased balance in the api but can't use the motion sensor. Is there plans to get this working?
> > > > Thanks!
> > > 
> > > 
> > > Hello, are you referring to the motion detect switch? Please try reloading the imou_life integration. If the issue persists, please check the Home Assistant logs for any abnormalities and provide them to us.
> > 
> > 
> > No, the switch is working perfect, but I need to have a sensor if "human detected" that is missing
> 
> Hello, human detected is currently not supported. We will consider supporting it in later versions

It would be great... or at least "motion detection" to make triggers with that.

Thanks!

**Comment by Imou-OpenPlatform** [comment]
**Update:** recent releases of this integration ship with an updated **`pyimouapi`** dependency that **adds motion-entity support for Ranger 2C 3MP** when the Open Platform reports motion for the device. Please **upgrade to the latest release**, **reload** the integration (or restart Home Assistant), and confirm you see the motion-related entities. If they are still missing or states look wrong **after** upgrading, paste your **Home Assistant version**, **integration version**, **model / productId**, **device serial number (S/N)**, and **log excerpts** from a period where motion should have fired—then we can treat it as a remaining bug or API payload issue.

**Comment by jlmadrugac** [comment]
Hi.
I have just upgraded the cam & the software and I can only enable motion detection or disable it... also it looks than it starts recording when motion is detected, but no "motion sensor" is available:
<img width="321" height="477" alt="Image" src="https://github.com/user-attachments/assets/c90ec022-873c-4899-8ff9-9e52446ddb54" />

**Comment by Imou-OpenPlatform** [comment]
> Hi. I have just upgraded the cam & the software and I can only enable motion detection or disable it... also it looks than it starts recording when motion is detected, but no "motion sensor" is available: <img alt="Image" width="321" height="477" src="https://private-user-images.githubusercontent.com/176721652/590598050-c90ec022-873c-4899-8ff9-9e52446ddb54.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Nzg1NTI5MzksIm5iZiI6MTc3ODU1MjYzOSwicGF0aCI6Ii8xNzY3MjE2NTIvNTkwNTk4MDUwLWM5MGVjMDIyLTg3M2MtNDg5OS04ZmY5LTllNTI0NDZkZGI1NC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUxMlQwMjIzNTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYmI3MmI1Mzk5YTZhYTg2YTBkZjRmMWY5MjRjZDI3MGRiYmM4M2Q2Y2I5NTVmZmFjZmI3ZGQ0MTJhNTlkNjA2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.Xvj8V2uuuG3lor0OHyzopEJ6w0pHx8niD5HnIvwKqjA">

I'm not sure I follow regarding 'no motion sensor available'. In this integration, Motion Detect is designed as a switch rather than a sensor.
If the issue persists, please provide a screenshot of your device page and the device information, along with the HA logs. This will help us investigate. Thanks.

**Comment by jlmadrugac** [comment]
If motion detection is designed as a switch, it is working now, thanks.
It would be great to have a sensor of "motion detection" so we can trigger actions when someone activates the trigger.

**Comment by Imou-OpenPlatform** [comment]
> If motion detection is designed as a switch, it is working now, thanks. It would be great to have a sensor of "motion detection" so we can trigger actions when someone activates the trigger.


I assume what you're looking for is for the device to trigger a detection event and notify Home Assistant (HA), which then performs the associated actions. We'll evaluate the feasibility of this in the future. Thanks for the feedback!

**Comment by Imou-OpenPlatform** [comment]
Closing this as answered.

**Summary**
- The **Motion detect switch** on Ranger 2C 3MP is the supported way to enable/disable detection (working on your side).
- For **automation triggers** when motion/human actually occurs, use **Event push** (Configure → enable event push, include **alarm**, external HA URL required). Events: `imou_life_event` / `imou_life_alarm` with `msg_type` such as `videoMotion` or `human`.
- A dedicated **binary sensor entity** for live motion/human state is **not** available in the current integration/API model; we are not tracking that as an open bug here.

If event push does not fire for your device after setup, please open a **new bug report** with integration version, diagnostics (`event_push.recent_msg_type_counts`), and logs from a motion event.

Thanks for the feedback.
